packages/peregrine/lib/Apollo/magentoGqlCacheLink.js

```diff
@@ -0,0 +1,59 @@
+import { ApolloLink } from '@apollo/client';
+import { BrowserPersistence } from '@magento/peregrine/lib/util';
+
+// The name of the header to exchange with the server.
+const CACHE_ID_HEADER = 'x-magento-cache-id';
+// The key in local storage where we save the cache id
+const LOCAL_STORAGE_KEY = 'magento_cache_id';
+const storage = new BrowserPersistence();
+
+/**
+ * The Magento GraphQL Cache Link class is an ApolloLink that is responsible for
+ * Venia sending the proper `x-magento-cache-id` header with each of its GraphQL requests.
+ */
+export default class MagentoGQLCacheLink extends ApolloLink {
+    // The links get reinstantiated on refresh.
+    // If we have an existing cache id value from a previous browsing session, use it.
+    #cacheId = storage.getItem(LOCAL_STORAGE_KEY) || null;
```

Comment:
Question: When does this get instantiated to re-check local storage?

Reply:
>When does this get instantiated to re-check local storage?

Every time the app is loaded: both on initial navigation to the site and on a browser refresh.

> And when does local storage get cleaned (if ever)?

I did not put any expiration date on this piece of data in local storage, it will stick around forever.